### DISCUSSION TITLE
Problem with opening new files when `:help` window is opened

### DISCUSSION THREAD

**Discussion by KriBielinski** [nvim-tree/nvim-tree.lua#3005] [Q&A] [answered]
### Description

While opening and closing files and `:help` buffer in a specific order, after then opening a new file using nvim-tree, the new file is open in a split window instead of replacing the `:help` buffer. 
The problem can be replicated using just the standard configuration.

### Neovim version

```text
NVIM v0.11.0-dev-1079+g7bf3a616e1                                                                                                   
Build type: Release                                                                                                                 
LuaJIT 2.1.1727870382
```

### Operating system and version

Distro: Linux Mint 21 Vanessa, base: Ubuntu 22.04 jammy

### nvim-tree version

28eac28

### Clean room replication

```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvt-min/site]])
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup({
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing nvim-tree and dependencies.")
  vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
vim.cmd([[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]])
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup({})
end

-- UNCOMMENT this block for diagnostics issues, substituting pattern and cmd as appropriate.
-- Requires diagnostics.enable = true in setup.
--[[
vim.api.nvim_create_autocmd("FileType", {
  pattern = "lua",
  callback = function()
    vim.lsp.start {
      name = "my-luals",
      cmd = { "lua-language-server" },
      root_dir = vim.loop.cwd(),
    }
  end,
})
]]
```

### Steps to reproduce

1. `nvim -nu /tmp/nvt-min.lua` to open Neovim
2. `:NvimTreeOpen` to open nvim-tree. It will appear in a split on the left.
3. Navigate to any file inside of nvim-tree and than press `Enter` to open it. The file will appear on the window to the right.
4. `:help` to open the help window. It should appear in a split window next to the opened file.
5. Navigate to the window with the file opened in step 3 (using `<C-w>+h/j/k/l`) and close that window with `:q`
6. Navigate to nvim-tree window (using `<C-w>+h/j/k/l`)
7. Open any file (doesn't have to be the same as before) as described in step 3

### Expected behavior

The newly opened file should replace the `:help` buffer (as it would do with any other buffer) but instead the new file openes in a vertical split next to the `:help` window. Also if instead of opening the file in step 7 you instead press `<C-x>` to attempt to open the file in horizontal split, the file still opens in vertical split.

### Actual behavior

The newly opened file opens in a vertical split (either when opening using `Enter` or `<C-x>`)

**Comment by alex-courtis** [comment]
Default vim behaviour is help splitting horizontally above all windows.

Have you got any other configuration that places the help window (I do https://github.com/alex-courtis/arch/blob/735510bdd95635e129ce34e072f9f0da4874a977/config/nvim/lua/amc/windows.lua#L144)

This doesn't address the issue though.

![20241116_120001](https://github.com/user-attachments/assets/ea7a3892-1a20-4c5d-b3ad-570c211f4fc8)

**Comment by alex-courtis** [comment]
nvim-tree won't open files in windows with "special" buffers.

See `:help nvim-tree.actions.open_file.window_picker.exclude`

```lua
            {
              filetype = {
                "notify",
                "packer",
                "qf",
                "diff",
                "fugitive",
                "fugitiveblame",
              },
              buftype = {
                "nofile",
                "terminal",
                "help",
              },
            }
```

This could be documented better.

**Comment by KriBielinski** [comment, marked as answer]
Hello @alex-courtis,

> nvim-tree won't open files in windows with "special" buffers.

Well, this doesn't seem to be exactly true as I can also create a sequence of actions like this:

1. `nvim -nu /tmp/nvt-min.lua`
2. `:help` to open help and then `<C-w>o` to make :help the only window
3. `:NvimTreeOpen` to open nvim-tree and open any file by navigating to it and using `Enter`

And now the newly opened file does actually replace the `:help` buffer.


But back to my original problem, like you have written in your last comment I have removed to help buffer from the list of excluded buffers by adding this to my nvim-tree configuration:
```
actions = {
  open_file = {
    window_picker = {
      exclude = {
        buftype = {
          "nofile",
          "terminal",
          --"help" -- remove :help from excluded list
        }
      },
    },
  },
}
```
and now everything works precisely as I expected it to. So thank you very much for help, I have solved the problem which I originally had at the beginning.

Well, since my problem is solved, should I close this issue now? I don't know, this my first issue ever on GitHub.

**Comment by alex-courtis** [comment]
> Well, since my problem is solved, should I close this issue now? I don't know, this my first issue ever on GitHub.

Many thanks for raising and please do continue.

I'm moving this to a Q/A discussion, so that others may benefit.

In the future, if you have a general question, please ask in there.
